assets/js/components/Dashboard/PaMessagesPage.tsx

```diff
@@ -42,7 +97,7 @@ const PaMessagesPage: ComponentType = () => {
               </Col> */}
             </Row>
             <Row>
-              <PaMessageTable paMessages={paMessages} />
+              {Array.isArray(data) && <PaMessageTable paMessages={data} />}
```

Comment:
It seems like we never actually narrow `data` from what I assume is a generic JSON value type returned from the fetcher. How does this work with the prop type needing to be a `PaMessage[]`?

Reply:
This "works" because things like `Response.prototype.json()` and `JSON.parse()` return `any`.

We could cast the value in the `usePaMessages` but that's effectively what happens we by the time the data is consumed in `<PaMessageTable />`

Getting type safe external data would require first parsing any external data to ensure it conforms the the types we're expecting. It would look something like this: https://github.com/mbta/screenplay/pull/433

Reply:
Huh. I kinda expected those to be properly typed (in which case we'd have gotten an error due to accessing possibly undefined properties on the data, and would have to explicitly type-cast it). [It's not like JSON is that complicated.](https://github.com/Microsoft/TypeScript/issues/15225#issuecomment-294718709) 🫠

I think I'd prefer an explicit type-cast somewhere close to where the data is fetched, but enforcing the type conformance would be even better.

Reply:
[Done!](https://github.com/mbta/screenplay/compare/75744043a2d6993fa135508113d577b85d850798..9fa47b418fd34e5fb7a4eee0c70417279c2a2cf7)